#! /bin/sh -e
## ubuntu-upstart.dpatch by Martin Pitt <mpitt@debian.org>
##
## DP: Use an upstart script on Ubuntu. Don't have it in debian/ by default,
## DP: since dh_installinit unconditionally prefers it over .init.

dpkg-vendor --is ubuntu || { echo -n '(ignored)' >&2; exit 0; }
. /usr/share/dpatch/dpatch-run

@DPATCH@
--- cups-1.4.4~/debian/cups.upstart	1970-01-01 01:00:00.000000000 +0100
+++ cups-1.4.4/debian/cups.upstart	2010-09-14 18:37:40.492181002 +0200
@@ -0,0 +1,40 @@
+# cups - CUPS Printing spooler and server
+
+description     "CUPS printing spooler/server"
+author          "Michael Sweet <msweet@apple.com>"
+
+start on (filesystem
+          and started dbus
+          and stopped udevtrigger)
+stop on runlevel [016]
+
+pre-start script
+    [ -x /usr/sbin/cupsd ]
+    
+    # load modules for parallel port support
+    if [ -r /etc/default/cups ]; then
+	. /etc/default/cups
+    fi
+    if [ "$LOAD_LP_MODULE" = "yes" -a -f /usr/lib/cups/backend/parallel \
+	 -a -f /proc/modules -a -x /sbin/modprobe ]; then
+	modprobe -q -b lp || true
+	modprobe -q -b ppdev || true
+	modprobe -q -b parport_pc || true
+    fi
+    
+    mkdir -p /var/run/cups/certs
+end script
+
+exec /usr/sbin/cupsd -f
+
+post-start script
+    # coldplug USB printers
+    if type udevadm > /dev/null 2>&1 && [ -x /lib/udev/udev-configure-printer ]; then
+        for printer in `udevadm trigger --verbose --dry-run --subsystem-match=usb \
+                --attr-match=bInterfaceClass=07 --attr-match=bInterfaceSubClass=01 2>/dev/null || true; \
+                        udevadm trigger --verbose --dry-run --subsystem-match=usb \
+                --sysname-match='lp[0-9]*' 2>/dev/null || true`; do
+            /lib/udev/udev-configure-printer add "${printer#/sys}"
+        done
+    fi
+end script
